### PR TITLE
Erl19 Breaks Mecking Callbacks/Behaviour Modules

### PR DESCRIPTION
**Pull request by RackerJohnMadrid**
This PR adds a single test with two additional helper modules used by the test.

The test demonstrates mocking both a behavior and callback module in the same
meck:new/? call.  This test passes in erlang 18.x as expected, but when running
from erlang 19, this test fails.

The error indicates that some compiler is not seeing the mocked behavior module
as the behavior module that it is.  I don't have a definitive reason for this,
but I have strong suspicions that the reason is related to:

http://erlang.org/pipermail/erlang-questions/2016-June/089615.html
http://erlang.org/pipermail/erlang-questions/2017-January/091396.html

Basically in erlang 19 the call M:module_info() seems to have stopped returning
the list of callbacks as attributes.  My guess is that Meck is using these in
some way to create the mocked version of the behavior module.  As the email
list posts state the callbacks are still there via beam_lib and potentially other ways.  It may be possible to fix this by using beam_lib, or perhaps by using merl to examine the AST.

### CODE REVIEW

**Comment by eproxus** [comment]
Inspecting the full error:

```erlang
{'EXIT',
    {{compile_forms,
         {error,[],
             [{"test/meck_callback_module.erl",
               [{2,erl_lint,{undefined_behaviour,meck_behaviour_module}}]}]}},
     [{meck_proc,start,
          [meck_callback_module,[no_link]],
          [{file,"src/meck_proc.erl"},{line,96}]},
      {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
      {meck,new,2,[{file,"src/meck.erl"},{line,201}]},
      {meck_behaviour_tests,'-mock_behaviour_module_test_/0-fun-0-',0,
          [{file,"test/meck_behaviour_tests.erl"},{line,9}]},
      {eunit_test,enter_context,4,[{file,"eunit_test.erl"},{line,324}]},
      {eunit_proc,run_group,2,[{file,"eunit_proc.erl"},{line,544}]},
      {eunit_proc,tests_inorder,3,[{file,"eunit_proc.erl"},{line,430}]},
      {eunit_proc,with_timeout,3,[{file,"eunit_proc.erl"},{line,320}]}]}}
```

You can get the same error (pretty printed) by removing the callback attribute from the behaviour module and then trying to compile the callback module.

What's happening here is that Meck tries to compile the behaviour module (but doesn't see the callback attribute, since it is removed) and just compiles the rest. Once the callback module is compiled by Meck, the behaviour attribute is used but that module no longer has any callbacks defined (because Meck didn't see them), thus the error.

**Comment by RackerJohnMadrid** [comment]
Yeah, we figured all that out, the poster of lastest mailing list post is a co-worker of mine that works on my team.  Just wanted to write a test that shows it here.

**Comment by eproxus** [comment]
Yeah, since I wasn't on the mailing list at that time I wrote a new email but haven't received a response yet...

http://erlang.org/pipermail/erlang-questions/2017-February/091690.html

**Comment by eproxus** [comment]
Closing due to inactivity.

**Comment by eproxus** [comment]
Closing due to inactivity.

**Comment by zzantozz** [comment]
@eproxus So is there anything to make this work, or is it just a rule now that you don't mock modules that have callbacks in them?

**Comment by eproxus** [comment]
@zzantozz For now that's the status quo unfortunately. I don't have a fix in mind which would solve this and no one replied on the mailing list so I'm kind of at a loss here.
